### PR TITLE
Self service css/images

### PR DESCRIPTION
**Pull request by brynyShift72**
[AB#7927](https://dev.azure.com/S72/fefacba9-96b6-4af2-a53d-050ed453e13a/_workitems/edit/7927) Supporting Self 

### Changes
- Added new site brand API.  Gets site image and CSS file URLs from the site brand endpoint in the user service.  Has separate feature toggles `self_service_site_images` and `self_service_css`.
- Removed `kibble/api/css.go` as the new site brand API makes it redundant.
- Added new site brand model.  Data from the site brand API is mapped onto this model and it is stored on the site model.  Has helper functions for easy usage in core-template.  Also added tests for this model.
- Changed render file writing function location from `kibble/render/translations_api_renderer.go` to `kibble/render/file_renderer.go`... I guess that makes it feel a bit more generic?
- Changed render function to GET and write an email logo to the build output if an email logo URL is present in the site brand model.  NOTE: Unfortunately, the email logo _must_ live in the build output as all one million of our email templates have hardcoded locations for this image.

### Site Model Change
- Removed CSSFilename property from the site model.  This property is no longer required thanks to the new site brand model but it unfortunately is a breaking change.  However, this property is not used by any clients in prod currently so IMO we can get away with it.

### Misc Changes
- Updated the golangci-lint github action because it was failing and wouldn't tell me why.  It works now.
- Added DS_Store to gitignore.
- Bumped `golang.org/x/sys` in go mod/sum because I couldn't build Kibble on my machine without doing that apparently?

### CODE REVIEW

File: kibble/api/site_brand.go
Comment (PeculiarGoat) [inline kibble/api/site_brand.go:63]: I reckon this repeated block can be extracted into a function.

File: kibble/models/site_brand.go
Comment (PeculiarGoat) [inline kibble/models/site_brand.go:21]: I think it's more accurate to describe as URL rather than path.

Comment (PeculiarGoat) [inline kibble/models/site_brand.go:21]: I think it's more accurate to describe as URL rather than path.

File: kibble/models/site_brand.go
Comment (PeculiarGoat) [inline kibble/models/site_brand.go:32]: This is another repeating block that can probably be extracted into a function.

Comment (PeculiarGoat) [inline kibble/models/site_brand.go:32]: This is another repeating block that can probably be extracted into a function.

File: kibble/api/site_brand.go
Comment (PeculiarGoat) [inline kibble/api/site_brand.go:23]: The other 'Load' functions are all PascalCase.

Comment (PeculiarGoat) [inline kibble/api/site_brand.go:23]: The other 'Load' functions are all PascalCase.

File: kibble/api/site_brand.go
Comment (PeculiarGoat) [inline kibble/api/site_brand.go:39]: That should be `siteBrands` for consistency right?

Comment (PeculiarGoat) [inline kibble/api/site_brand.go:39]: That should be `siteBrands` for consistency right?

File: kibble/api/site_brand.go
Comment (PeculiarGoat) [inline kibble/api/site_brand.go:56]: `Info` should be lowercase `info`.

File: kibble/api/site_brand.go
Comment (PeculiarGoat) [inline kibble/api/site_brand.go:50]: I think this is a bit overkill.  We don't need a print out of all the data we've grabbed from the API... we just want to know if the feature is on or off.

```
if site.Toggles[feature_key] {
  fmt.Println("Feature Toggle \"%s\" enabled.  Using Cloudfront URLs where possible.", feature_key)
} else {
  fmt.Println("Feature Toggle \"%s\" disabled.  Using template default URLs.", feature_key)
}
```

Something like that.

Comment (PeculiarGoat) [inline kibble/api/site_brand.go:50]: I think this is a bit overkill.  We don't need a print out of all the data we've grabbed from the API... we just want to know if the feature is on or off.

```
if site.Toggles[feature_key] {
  fmt.Println("Feature Toggle \"%s\" enabled.  Using Cloudfront URLs where possible.", feature_key)
} else {
  fmt.Println("Feature Toggle \"%s\" disabled.  Using template default URLs.", feature_key)
}
```

Something like that.

File: kibble/models/site_brand_test.go
Comment (PeculiarGoat) [inline kibble/models/site_brand_test.go:22]: Could probably improve the test name a bit.  Something like `TestGetImageWhenSiteBrandImagesIsEmpty`.  Same for the others below.

Comment (PeculiarGoat) [inline kibble/models/site_brand_test.go:22]: Could probably improve the test name a bit.  Something like `TestGetImageWhenSiteBrandImagesIsEmpty`.  Same for the others below.

File: kibble/render/render.go
Comment (PeculiarGoat) [inline kibble/render/render.go:83]: Remove this empty line!

Comment (PeculiarGoat) [inline kibble/render/render.go:83]: Remove this empty line!

File: kibble/render/render.go
Comment (PeculiarGoat) [inline kibble/render/render.go:107]: Would be tidier to do `emailLogoURL := site.SiteBrand["email-logo"]` here... no need to write out the extra characters with the Get function in Kibble.

Comment (PeculiarGoat) [inline kibble/render/render.go:107]: Would be tidier to do `emailLogoURL := site.SiteBrand["email-logo"]` here... no need to write out the extra characters with the Get function in Kibble.

Comment (PeculiarGoat) [reply kibble/render/render.go:107]: Also, this block could be extracted into it's own function given that it has no impact on the rest of the render function and the render function is pretty big and ugly now.

Comment (PeculiarGoat) [reply kibble/render/render.go:107]: Also, this block could be extracted into it's own function given that it has no impact on the rest of the render function and the render function is pretty big and ugly now.

File: kibble/models/site_brand_test.go
Comment (brynyShift72) [reply kibble/models/site_brand_test.go:40]: Yeah I thought that was the case, however I did just make the tests match the ACs

File: kibble/api/site_brand.go
Comment (brynyShift72) [reply kibble/api/site_brand.go:50]: I'm in two minds about it, yes it looks less busy, but I do like to know what asset types are being pulled through, similar to translations, we print the languages that is enabled for each client.

Comment (PeculiarGoat) [reply kibble/api/site_brand.go:50]: Well even if you do decide to print out all of the image types, you could do it much simpler without a string builder being passed through conditional and loop blocks.

File: kibble/models/site_brand.go
Comment (PeculiarGoat) [inline kibble/models/site_brand.go:32]: Given that you have called the key `assetType`, I think `assetMap` (or something similar) would be a more suitable name for `brandType` here.

Comment (PeculiarGoat) [reply kibble/models/site_brand.go:32]: I also think that `defaultPath` should be renamed to `defaultURL` in this file.

File: kibble/models/site_brand_test.go
Comment (PeculiarGoat) [reply kibble/models/site_brand_test.go]: I reckon remove them.  The unit tests don't need to match the ACs because the bit of code you are testing here has no knowledge of the ACs or the different image types... you only need to test that it returns the default if nothing is found, or returns the URL from the map if it is found... doesn't matter what the data actually is.

**Review by stajs — COMMENTED** [review]
Will review properly once out of draft and post-1.0.

File: kibble/models/site_brand_test.go
Comment (brynyShift72) [reply kibble/models/site_brand_test.go]: I left the two top ones testing completely empty info there, as I think having a Site variable with no initialized data, but still showing the function to be working properly is nice, but can remove, I have removed all tests after the test that starts on L58

File: kibble/api/site_brand.go
Comment (PeculiarGoat) [inline kibble/api/site_brand.go:46]: The toggle variables can be moved to the top of the function and used in that first early exit.

File: kibble/api/site_brand.go
Comment (PeculiarGoat) [inline kibble/api/site_brand.go:49]: This doesn't feel right... the `self_service_css` toggle probably shouldn't dictate whether or not we use _any_ links.  I would prefer to only use the toggle to decide on the CSS link, and allow all other links to be added regardless.

File: kibble/models/site_brand.go
Comment (PeculiarGoat) [inline kibble/models/site_brand.go:21]: I think this comment would make more sense over the `getItem` function.

File: kibble/models/site_brand.go
Comment (PeculiarGoat) [inline kibble/models/site_brand.go:27]: Search and replace `path` => `URL`!

**Comment by PeculiarGoat** [comment]
I'm happy to approve once those last few comments have been addressed.  Better wait on @stajs approval before merge though.

File: kibble/api/site_brand.go
Comment (brynyShift72) [reply kibble/api/site_brand.go:49]: But hasn't the API intertwined these two ideas?

**Review by PeculiarGoat — APPROVED** [review]
Nice work!  👍 
Better wait for a thumbs up from @stajs too!

**Review by l0ud0gg — APPROVED** [review]
i'll review once its not in draft

**Comment by PeculiarGoat** [comment]
Does this need to be blocked?

**Comment by PeculiarGoat** [comment]
The CSS GET endpoint also had a breaking change though so there is currently no released version of Kibble that can handle  CSS files.  However, the only client in production with the feature turned on is the test client so not a big deal I guess.

**Review by PeculiarGoat — APPROVED** [review]
I'm happy with this PR.  Can we get an approval from another Kibbler too?  @additiverich ?

**Comment by PeculiarGoat** [comment]
I've updated the PR description a bit.  Could I please get some other eyes on this @additiverich @matthewthomsonnz @gscragg !

**Review by gscragg — APPROVED** [review]
Nice work.

**Comment by PeculiarGoat** [comment]
> Nice work.

I'll let Bryny know!

**Pull request merged**
